scm_v3c/optical.c

```diff
@@ -35,18 +51,46 @@ optical_vars_t optical_vars;
 void optical_init(void) {
     
     memset(&optical_vars, 0, sizeof(optical_vars_t));
-    
-    // Target radio LO freq = 2.4025G
+
     // Divide ratio is currently 480*2
     // Calibration counts for 100ms
-    optical_vars.LC_target  = REFERENCE_LC_TARGET;
-    optical_vars.LC_code    = DEFUALT_INIT_LC_CODE;
+    optical_vars.LC_target = REFERENCE_LC_TARGET;
+    optical_vars.LC_coarse = DEFAULT_INIT_LC_COARSE;
+    optical_vars.LC_mid = DEFAULT_INIT_LC_MID;
+    optical_vars.LC_fine = DEFAULT_INIT_LC_FINE;
 }
 
-uint8_t optical_getCalibrationFinshed(void) {
+void optical_enableLCCalibration(void) {
+    optical_vars.optical_LC_cal_enable = true;
+    optical_vars.optical_LC_cal_finished = false;
+
+    optical_vars.cal_LC_coarse = LC_CAL_COARSE_MIN;
+    optical_vars.cal_LC_mid = LC_CAL_MID_MIN;
+    optical_vars.cal_LC_fine = LC_CAL_FINE_MIN;
+    optical_vars.cal_LC_diff = 0xFFFFFFFF;
+    LC_FREQCHANGE(optical_vars.cal_LC_coarse, optical_vars.cal_LC_mid, optical_vars.cal_LC_fine);
+}
+
+bool optical_getCalibrationFinished(void) {
```

Comment:
Oh I see there is a typo of this function that you fixed, I guess all other projects are using the wrong old name.

Reply:
I will fix this in the other projects.

Reply:
https://github.com/PisterLab/scum-test-code/pull/17/commits/c5d0f48ebf31b7a17eb5ca83e68687f3e9fb9cb5

Reply:
Great!